only long running recipes (arcs) allowed to specify Ids for `create` fate handles

mmandlis opened the pull request.

piotrswigon commented on src/runtime/recipe/recipe.ts:
```diff
@@ -697,6 +698,16 @@ export class Recipe implements Cloneable<Recipe> {
     return slot;
   }
+
+  get isLongRunning(): boolean {
```
Maybe we could have ephemeral vs. persistent if we have to put something in code? That's a better name IMO than long-running. And with stores of long-runnning arcs being instantiated eagerly, they are kind of like "persistent" arcs. WDYT? @shans

mmandlis replied:
I'm happy to rename in general, but "persistent" would be confusing, because `persistent` capability means on disk, but long running arcs could use in-memory storage as well.

shans commented:
There's no reason to prevent short-running arcs from providing ids. They'll end up with storage keys that incorporate their arc IDs same as long-running arcs.

mmandlis commented:
> There's no reason to prevent short-running arcs from providing ids. They'll end up with storage keys that incorporate their arc IDs same as long-running arcs.

in the TS runtime, where there is `Runtime` object and planning, we use a map of stores by id from all arcs to resolved mapped recipe handles. so Id must be unique. otherwise running multiple ephemeral arcs with the same recipe will cause a collision.

i could update these ids from the one set in the recipe to `${arcId}${recipeHandleId}` instead, is that better?

mmandlis commented:
> > There's no reason to prevent short-running arcs from providing ids. They'll end up with storage keys that incorporate their arc IDs same as long-running arcs.
> 
> in the TS runtime, where there is `Runtime` object and planning, we use a map of stores by id from all arcs to resolved mapped recipe handles. so Id must be unique. otherwise running multiple ephemeral arcs with the same recipe will cause a collision.
> 
> i could update these ids from the one set in the recipe to `${arcId}${recipeHandleId}` instead, is that better?

as discussed offline, i removed the validity check, only leaving the `isLongRunning` arc check method and updated tests. @shans  PTAL

mmandlis commented:
merging, as Alex could use this method. if there are further comments, will address as a follow up.
